### ISSUE TITLE
Idea: V4 - Support formatting changed files

### ISSUE THREAD

**Issue by lukeapage** [prettier/prettier#15964]
I'm pretty excited by the performance improvements in v4, but I'd also loved a changedSince argument (similiar to support in jest) which uses your source control provider to see what files are changed and then only run on these files.

I use a fork of pretty-quick: https://github.com/prettier/pretty-quick

even with parallel prettier it can take 2 minutes to check our codebase, so the v4 benefits (although great) don't make it workable.

I could possibly contribute this if it is accepted.

**Comment by sosukesuzuki** [comment]
Is [cache](https://prettier.io/docs/en/cli#--cache) not enough?

**Comment by lukeapage** [comment]
No, we have 22k thousand prettier'd files in the repo and the numbers I quotes are with a cache.

But I haven't collected numbers for v4 - Its possible the v4 cache is much much faster? It would help to have a alpha published with the latest release changes

**Comment by lukeapage** [comment]
@sosukesuzuki 

The new v4 is much faster.. v3 took around 3 minutes. v4 - cold is 1m10s and warm is 25s. but with changed since it is a few seconds...

Would you accept a PR for this feature?
